tasks/collector.py

```diff
@@ -242,3 +258,215 @@ def generate(ctx):
                     f.write(content)
 
                 print(f"Updated package name and ensured license header in: {file_path}")
+
+
+GITHUB_API_URL = "https://api.github.com/repos"
+
+
+def fetch_latest_release(repo):
+    url = f"{GITHUB_API_URL}/{repo}/releases/latest"
+    print(f"Fetching latest release from {url} for {repo}")
+    response = requests.get(url)
+    if response.status_code == 200:
+        data = response.json()
+        return data["tag_name"]
+    else:
+        return None
+
+
+def fetch_core_module_versions(version):
+    """
+    Fetch versions.yaml from the provided URL and build a map of modules with their versions.
+    """
+    url = f"https://raw.githubusercontent.com/open-telemetry/opentelemetry-collector/{version}/versions.yaml"
+    print(f"Fetching versions from {url}")
+
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raises an HTTPError if the HTTP request returned an unsuccessful status code
+    except requests.exceptions.RequestException as e:
+        raise Exit(
+            color_message(f"Failed to fetch the YAML file: {e}", Color.RED),
+            code=1,
+        ) from e
+
+    yaml_content = response.content
+
+    try:
+        data = yaml.safe_load(yaml_content)
+    except yaml.YAMLError as e:
+        raise Exit(
+            color_message(f"Failed to parse YAML content: {e}", Color.RED),
+            code=1,
+        ) from e
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** You should consider using the github API instead of requests (`tasks.libs.ciproviders.github_api`)‏

Reply:
github_api only provides method download_from_url  , that might be more complex than existing code